vm/src/obj/objint.rs

```diff
@@ -311,18 +341,18 @@ impl PyInt {
             return Ok(vm.ctx.not_implemented());
         }
 
-        if let Some(n_bits) = get_value(&other).to_usize() {
-            return Ok(vm.ctx.new_int((&self.value) << n_bits));
-        }
+        let other = other.payload::<PyInt>().unwrap();
+        inner_lshift(self, other, vm)
+    }
 
-        // i2 failed `to_usize()` conversion
-        match get_value(&other) {
-            v if *v < BigInt::zero() => Err(vm.new_value_error("negative shift count".to_string())),
-            v if *v > BigInt::from(usize::max_value()) => {
-                Err(vm.new_overflow_error("the number is too large to convert to int".to_string()))
-            }
-            _ => panic!("Failed converting {} to rust usize", get_value(&other)),
+    #[pymethod(name = "__rlshift__")]
+    fn rlshift(&self, other: PyObjectRef, vm: &VirtualMachine) -> PyResult {
+        if !objtype::isinstance(&other, &vm.ctx.int_type()) {
+            return Ok(vm.ctx.not_implemented());
         }
+
+        let other = other.payload::<PyInt>().unwrap();
```

Comment:
It is better to not directly access the payload attribute. Use `get_value` instead.

Reply:
I found some similar problems in other int methods. I suggest this change to another PR.